src/ChannelAllocationChromosomeMutation.java

```diff
@@ -0,0 +1,41 @@
+public class ChannelAllocationChromosomeMutation implements IMutationAlgorithm<ChannelAllocationChromosome> {
+
+    public ChannelAllocationChromosome mutate(ChannelAllocationChromosome chromosome, double geneMutationProbability, boolean uniform){
+        float Pm = RandomGenerator.rand();
```

Comment:
What is this variable for ? I see you are comparing it to do mutation but you should compare against the passed variable `geneMutationProbability`